localstack/services/awslambda/lambda_executors.py

```diff
@@ -125,14 +127,34 @@ def execute(self, func_arn, func_details, event, context=None, version=None, asy
                 (taskdir, LAMBDA_EXECUTOR_CLASS, handler, LAMBDA_EVENT_FILE))
 
         # determine the command to be executed (implemented by subclasses)
-        cmd = self.prepare_execution(func_arn, environment, runtime, command, handler, lambda_cwd)
+        cmd = self.prepare_execution(func_arn, func_details, environment, runtime, command, handler, lambda_cwd)
 
         # lambci writes the Lambda result to stdout and logs to stderr, fetch it from there!
         LOG.debug('Running lambda cmd: %s' % cmd)
         result, log_output = self.run_lambda_executor(cmd, environment, async)
         LOG.debug('Lambda result / log output:\n%s\n>%s' % (result.strip(), log_output.strip().replace('\n', '\n> ')))
         return result, log_output
 
+    def lambda_docker_network(self, func_arn, func_details):
+        network = None
+
+        if config.LAMBDA_SUBNET_AS_DOCKERNET and func_details.vpc_config:
+            subnets = func_details.vpc_config.get('SubnetIds', [])
+            if len(subnets) > 0:
+                network = subnets[0]
+
+        if not network and config.LAMBDA_DEFAULT_DOCKER_NETWORK:
+            network = config.LAMBDA_DEFAULT_DOCKER_NETWORK
+
+        return network
+
+    def lambda_docker_cmd_networksection(self, func_arn, func_details):
+        network = self.lambda_docker_network(func_arn, func_details)
+        if network:
+            return ' --network="%s" ' % network
```

Comment:
Nitpick: can we change this to `return ' --network="%s" ' % network if network else ""`

Reply:
You can change whatever you want here.

This PR is over 6 months old, with no signs of it ever getting merged.  

I no longer use local stack as it was too brittle for my needs, and have moved on.  

I will no longer be devoting any time or attention to this PR, or to local stack in general.

Sorry it didn't work out.

Reply:
@gadgetjunkie what did you end up moving to?

Reply:
@cabdesigns I am simply using separate CloudFormation stacks in AWS for each environment/developer.  It costs more, but it eliminates the many performance and inconsistency issues we were seeing when we tried to run our stuff in localstack.